### DISCUSSION TITLE
Choose one of the optimal outcomes based on certain condition in CP-SAT

### DISCUSSION THREAD

**Discussion by arnabadas** [google/or-tools#3713] [CP-SAT questions]
**What language and solver does this apply to?**
All / CP-SAT

**Describe the problem you are trying to solve.**
I am trying to optimize two objectives. But I can only provide one single objective. My problem doesn't have one single solution, but multiple possible optimal solutions.
**Describe the solution you'd like**
I heard CP-SAT calculates the optimal solution as feasible multiple times with different constraint RHS values in order to prove optimality. I would like to choose one of those optimal solutions where a specific equation of decision variables (it may also be the LHS of a constraint) has the minimum/maximum RHS value. It will seem like CP-SAT can solve for two objectives.
**Describe alternatives you've considered**
I have managed to solve my problem.
**Additional context**

**Comment by lperron** [comment]
You misheard.

The solver searches for improving solutions, until infeasibility is proved, meaning the last one was optimal.

You can search for the optimal solution, record the value, then remove the objective, add the constraint obj == optimal_value, then use the parameters enumerate_all_solutions=true and resolve.

Furthermore, I do not understand the connection between muliple optimal objectives, and the fact that you have 2 different objectives you want to combine/explore.
